gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/ChangeMonitorUtils.java

```diff
@@ -28,20 +30,24 @@ private ChangeMonitorUtils() {
   }
 
   /**
-   * Performs checks for duplicate messages and heartbeat operation prior to processing a message. Returns true if
-   * the pre-conditions above don't apply and we should proceed processing the change event
+   * Performs checks for duplicate messages, heartbeat message types, or null dag action types all of which cannot or
+   * should not be processed. Returns true if the pre-conditions above don't apply, and we should proceed processing
+   * the change event
    */
-  public static boolean shouldProcessMessage(String changeIdentifier, LoadingCache<String, String> cache,
-      String operation, String timestamp) {
+  public static boolean isValidAndUniqueMessage(String changeIdentifier, LoadingCache<String, String> cache,
+      String operation, String timestamp, ContextAwareMeter duplicateMessagesMeter,
+      ContextAwareMeter heartbeatMessagesMeter) {
```

Comment:
nit: I would put all the elements from the message together as the first params, and then the cache and the meters